### PR TITLE
Create Flickswitch status check request

### PR DESCRIPTION
**Pull request by Lindani**

### CODE REVIEW

**Comment by Lindani** [comment]
@gsvr : Please review

**Comment by Lindani** [comment]
@gsvr : Please re-review

File: recharges/tasks.py [hunk 1]
```diff
@@ -167,4 +224,37 @@ def run(self, recharge_id, **kwargs):
...
+    def run(self, recharge_id, **kwargs):
+        # Call get_recharge() by recharge_id to get recharge object
+        # Call get_hotsocket_status()
+        # by supplying hotsocket reference and and ref_id to get status code
+        # Check if hotsocket status code is 000
+        # Recharge status code should be 2 and such status should be saved
+        # Return "Recharge for " + recharge.msisdn + " is successful"
+        return "recharge is succesful"
+
+hotsocket_status = Check_Hotsocket_Status()
```
Comment (gsvr) [inline recharges/tasks.py]: Naming is inconsistent.

File: setup.cfg [hunk 2]
```diff
@@ -4,4 +4,4 @@ exclude = */migrations/*.py,*/manage.py,local_settings.py,.git
 
 [pytest]
 python_files=test*.py
-addopts = --verbose --cov . --ds=gopherairtime.testsettings
+addopts = --verbose --ds=gopherairtime.testsettings
```
Comment (gsvr) [inline setup.cfg]: This should not have been committed - undo the change.

File: recharges/tasks.py [hunk 3]
```diff
@@ -167,4 +224,37 @@ def run(self, recharge_id, **kwargs):
...
+class Check_Hotsocket_Status(Task):
+
+    """
+    Task to check hotsocket recharge request and sets the recharge model
+    status to successful if the airtime has been loaded to the user's phone.
+    """
+    name = "recharges.tasks.Check_Hotsocket_Status"
+
+    def prep_hotsocket_status_dict(self, recharge_id, hs_reference):
+
```
Comment (gsvr) [inline recharges/tasks.py]: Docstring needed.

File: recharges/tasks.py [hunk 4]
```diff
@@ -167,4 +224,37 @@ def run(self, recharge_id, **kwargs):
...
+        hotsocket_data = {
+            'username': settings.HOTSOCKET_API_USERNAME,
+            'as_json': True,
+            'token': get_token(),
+            'reference': recharge_id + settings.HOTSOCKET_REFBASE,
+            'hotsocket_ref': hs_reference
```
Comment (gsvr) [inline recharges/tasks.py]: From the documentation:
`Status can handle your reference or the reference HOTSOCKET returned - just
leave the other reference blank (it’s XOR, so don’t send both, you will get an error message).`

File: recharges/tests.py [hunk 5]
```diff
@@ -376,3 +456,17 @@ def test_hotsocket_get_airtime_unrecoverable(self):
         # Check
         self.assertEqual(result.get(),
                          "airtime request for +277244555 is unrecoverable")
+
+    def test_Check_Hotsocket_Status(self):
```
Comment (gsvr) [inline recharges/tests.py]: Don't use capital letters here.

File: recharges/tests.py [hunk 6]
```diff
@@ -376,3 +456,17 @@ def test_hotsocket_get_airtime_unrecoverable(self):
...
+        recharge_id = self.make_recharge()
+        result = hotsocket_status(recharge_id)
+        self.assertEqual(result, "recharge is succesful")
```
Comment (gsvr) [inline recharges/tests.py]: spelling: 'successful'

File: recharges/tests.py [hunk 7]
```diff
@@ -376,3 +456,17 @@ def test_hotsocket_get_airtime_unrecoverable(self):
...
+
+
+class TestRechrgeStatus(TaskTestCase):
+    def test_prep_hotsocket_status_dict(self):
+        self.make_account()
+        result = hotsocket_status.prep_hotsocket_status_dict(recharge_id=1003,
+                                                             hs_reference=4507)
+        self.assertEqual(result['reference'], 11003)
+        self.assertEqual(result['hotsocket_ref'], 4507)
```
Comment (gsvr) [inline recharges/tests.py]: You should also be checking for data that you haven't passed through, like the `token` and `username`.

File: recharges/tests.py [hunk 8]
```diff
@@ -376,3 +456,17 @@ def test_hotsocket_get_airtime_unrecoverable(self):
...
                          "airtime request for +277244555 is unrecoverable")
+
+    def test_Check_Hotsocket_Status(self):
+        recharge_id = self.make_recharge()
+        result = hotsocket_status(recharge_id)
+        self.assertEqual(result, "recharge is succesful")
+
+
+class TestRechrgeStatus(TaskTestCase):
+    def test_prep_hotsocket_status_dict(self):
```
Comment (gsvr) [inline recharges/tests.py]: Move this test function into the previously used test class TestRechargeFunctions for consistency.

**Comment by gsvr** [comment]
@imsickofmaps Please review. I was a bit rushed so I expect comments.

File: recharges/tasks.py [hunk 9]
```diff
@@ -252,3 +252,79 @@ def run(self, recharge_id, **kwargs):
...
+        return recharge_status_post.json()
+
+    def run(self, recharge_id, **kwargs):
+        l = self.get_logger(**kwargs)
+        l.info("Looking up Hotsocket status")
+        hs_status = self.request_hotsocket_status(recharge_id)
+        hs_status_code = hs_status["response"]["status"]
+
+        if hs_status_code == "0000":
+            # recharge successful
```
Comment (imsickofmaps) [inline recharges/tasks.py]: this just means lookup was successful - not the actual recharge - see section 11.4 of the guide

**Comment by imsickofmaps** [comment]
@gsvr review my changes please

**Comment by gsvr** [comment]
@imsickofmaps Just a spelling mistake, :+1: otherwise.
